osde2e/route_monitor_operator_tests.go

```diff
@@ -16,68 +16,85 @@ import (
 )
 
 var _ = Describe("Route Monitor Operator", Ordered, func() {
 	var (
-		k8s      *openshift.Client
-		err      error
-		pod      *corev1.Pod
-		svc      *corev1.Service
-		appRoute *routev1.Route
-		rmo      *routemonitorv1alpha1.RouteMonitor
+		k8s *openshift.Client
 	)
 	const (
-		operatorNamespace            = "openshift-route-monitor-operator"
-		deploymentName               = "route-monitor-operator-controller-manager"
-		operatorName                 = "route-monitor-operator"
-		consoleName                  = "console"
-		pollingDuration              = 10 * time.Minute
-		routeMonitorName             = "routemonitor-e2e-test"
-		defaultDesiredReplicas int32 = 1
+		namespace        = "openshift-route-monitor-operator"
+		deploymentName   = "route-monitor-operator-controller-manager"
+		operatorName     = "route-monitor-operator"
+		consoleName      = "console"
+		pollingDuration  = 10 * time.Minute
+		routeMonitorName = "routemonitor-e2e-test"
 	)
 	BeforeAll(func(ctx context.Context) {
 		log.SetLogger(GinkgoLogr)
+		var err error
 		k8s, err = openshift.New(GinkgoLogr)
 		Expect(err).ShouldNot(HaveOccurred(), "unable to setup k8s client")
 		Expect(routemonitorv1alpha1.AddToScheme(k8s.GetScheme())).Should(Succeed(), "unable to register RouteMonitor scheme")
 	})
 
 	It("is installed", func(ctx context.Context) {
+		By("checking the namespace exists")
+		err := k8s.Get(ctx, namespace, "", &corev1.Namespace{})
+		Expect(err).ShouldNot(HaveOccurred(), "namespace %s not found", namespace)
+
 		By("checking the deployment exists and is available")
-		EventuallyDeployment(ctx, k8s, deploymentName, operatorNamespace).Should(BeAvailable())
+		EventuallyDeployment(ctx, k8s, deploymentName, namespace).Should(BeAvailable())
 	})
 
 	It("can be upgraded", func(ctx context.Context) {
 		By("forcing operator upgrade")
-		err = k8s.UpgradeOperator(ctx, operatorName, operatorNamespace)
-		Expect(err).NotTo(HaveOccurred(), "operator upgrade failed")
+		err := k8s.UpgradeOperator(ctx, operatorName, namespace)
+		Expect(err).ShouldNot(HaveOccurred(), "operator upgrade failed")
 	})
 
 	It("has all of the required resources", func(ctx context.Context) {
-		promclient, err := prometheusop.NewForConfig(k8s.GetConfig())
-		Expect(err).ShouldNot(HaveOccurred(), "failed to configure Prometheus-operator clientset")
-		_, err = promclient.MonitoringV1().ServiceMonitors(operatorNamespace).Get(ctx, consoleName, metav1.GetOptions{})
-		Expect(err).NotTo(HaveOccurred(), "Could not get console serviceMonitor")
-		_, err = promclient.MonitoringV1().PrometheusRules(operatorNamespace).Get(ctx, consoleName, metav1.GetOptions{})
-		Expect(err).NotTo(HaveOccurred(), "Could not get console prometheusRule")
+		dynamicClient, err := dynamic.NewForConfig(k8s.GetConfig())
+		Expect(err).ShouldNot(HaveOccurred(), "failed creating the dynamic client: %w", err)
+		serviceMonitorsClient := dynamicClient.Resource(schema.GroupVersionResource{
```

Comment:
you can create these in the `BeforeAll` and have them be global so you don't have to create them twice